website/server/controllers/api-v4/members.js

```diff
@@ -0,0 +1,34 @@
+import {model as User} from '../../models/user';
+
+let api = {};
+
+
+api.getUsernameAutocompletes = {
+  method: 'GET',
+  url: '/members/find/:username',
+  middlewares: [],
+  async handler (req, res) {
+    req.checkParams('username', res.t('invalidReqParams')).notEmpty();
+
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let username = req.params.username.toLowerCase();
+    if (username[0] === '@') username = username.slice(1, username.length);
+
+    if (username.length < 1) {
+      res.respond(200, []);
+      return;
+    }
+
+    let members = await User
+      .find({'auth.local.lowerCaseUsername': {$regex: `.*${username}.*`}, 'flags.verifiedUsername': true, 'preferences.searchableUsername': {$ne: false}})
```

Comment:
let's make sure we have the necessary indexes here or this'll be super slow

Reply:
tbh I'm a bit worried about the number of requests this route will get and the performances, considering the $regex type of search. Can we at least make it so the prefix part is fixed so that the search is faster as explained here https://docs.mongodb.com/manual/reference/operator/query/regex/#index-use?

Reply:
I'd say it's fine for now, if people complain or there are no performance issues we can extend it but I'd prefer to start with the faster version if possible

Reply:
Hey @phillipthelen looks like this wasn't actually changed, the regexp

> A regular expression is a “prefix expression” if it starts with a caret (^) or a left anchor (\A), followed by a string of simple symbols. For example, the regex /^abc.*/ will be optimized by matching only against the values from the index that start with abc.

Reply:
Marked as unresolved

Reply:
Sorry, I thought we'd leave the regex as is for now, since you last comment was it's fine for now. I changed it to work the way you requested